kuiper-game/Domain/CelestialBodies/CelestialBody.cs

```diff
@@ -1,151 +1,76 @@
         {
             const double VELOCITYCONSTANT = 29.8;
             var velocity = VELOCITYCONSTANT / (orbitRadius / Math.Sqrt(orbitRadius));
 
-            return Create(name, orbitRadius, velocity, originDegrees, parent, bodyType);
+            return Create<T>(name, orbitRadius, velocity, originDegrees, parent);
         }
 
-        public static CelestialBody Create(string name, double orbitRadius, double velocity, 
-            double originDegrees, CelestialBody parent, CelestialBodyType bodyType)
+        public static T Create<T>(string name, double orbitRadius, double velocity, double originDegrees, CelestialBody parent)
+            where T : CelestialBody
         {
-            CelestialBody body = null;
-
-            switch (bodyType)
-            {
-                case CelestialBodyType.Star:
-                    body = create<Star>(name, orbitRadius, velocity, originDegrees, null);
-                    break;
-                case CelestialBodyType.Planet:
-                    body = create<Planet>(name, orbitRadius, velocity, originDegrees, parent);
-                    break;
-                case CelestialBodyType.Moon:
-                    body = create<Moon>(name, orbitRadius, velocity, originDegrees, parent);
-                    break;
-                case CelestialBodyType.GasGiant:
-                    body = create<GasGiant>(name, orbitRadius, velocity, originDegrees, parent);
-                    break;
-                case CelestialBodyType.DwarfPlanet:
-                    body = create<DwarfPlanet>(name, orbitRadius, velocity, originDegrees, parent);
-                    break;
-                case CelestialBodyType.Asteroid:
-                    body = create<Asteroid>(name, orbitRadius, velocity, originDegrees, parent);
-                    break;
-            }
-
-            if (body.parent != null)
-            {
-                parent.AddSatellite(body);
-            }
-            return body;
+            return (T)Create(typeof(T), name, orbitRadius, velocity, originDegrees, parent);
         }
-
-        public void AddSatellite(CelestialBody satellite)
-        {
-            satellites.Add(satellite);
-        }
-
-        private static CelestialBody create<T>(string name, double orbitRadius, double velocity,
-            double originDegrees, CelestialBody parent) where T : CelestialBody, new()
+        
+        public static CelestialBody Create(Type celestialType, string name, double orbitRadius, double velocity, double originDegrees, CelestialBody parent)
         {
-            var body = new T()
-            {
-                name = name,
-                orbitRadius = orbitRadius,
-                velocity = velocity,
-                originDegrees = originDegrees,
-                satellites = new List<CelestialBody>(),
-                parent = parent
-            };
 
+            var body = (CelestialBody)Activator.CreateInstance(celestialType,
```

Comment:
I believe new T to be the more idiomatic way of expressing it (although it does technically call Activator.CreateInstance behind the scenes.

It is however just as fast.

If you really want you could do something like 
`Expression.New(typeof(T))).Compile();` which is faster, by quite a margin.